python/servo/post_build_commands.py

```diff
@@ -211,10 +209,13 @@ def rr_replay(self):
         'params', nargs='...',
         help="Command-line arguments to be passed through to cargo doc")
     def doc(self, params):
-        self.ensure_bootstrapped()
-        if not path.exists(path.join(self.config["tools"]["rust-root"], "doc")):
-            Registrar.dispatch("bootstrap-rust-docs", context=self.context)
-        rust_docs = path.join(self.config["tools"]["rust-root"], "doc")
+        env = os.environ.copy()
+        env["RUSTUP_TOOLCHAIN"] = self.toolchain()
+        rustc_path = check_output(["rustup" + BIN_SUFFIX, "which", "rustc"], env=env)
+        assert path.basename(path.dirname(rustc_path)) == "bin"
+        toolchain_path = path.dirname(path.dirname(rustc_path))
+        rust_docs = path.join(toolchain_path, "share", "doc", "rust", "html")
```

Comment:
My only question is whether `./mach cargo-gecko doc` behaves as expected with these changes, since we use it in upload_docs.sh.